src/app/main.cpp

```diff
@@ -837,25 +845,17 @@ int main( int argc, char *argv[] )
 
   // TODO: use QgsSettings
   QSettings *customizationsettings = nullptr;
-  if ( !optionpath.isEmpty() || !configpath.isEmpty() )
-  {
-    // tell QgsSettings to use INI format and save the file in custom config path
-    QSettings::setDefaultFormat( QSettings::IniFormat );
-    QString path = optionpath.isEmpty() ? configpath : optionpath;
-    QSettings::setPath( QSettings::IniFormat, QSettings::UserScope, path );
-    customizationsettings = new QSettings( QSettings::IniFormat, QSettings::UserScope, QStringLiteral( "QGIS" ), QStringLiteral( "QGISCUSTOMIZATION2" ) );
-  }
-  else
-  {
-    customizationsettings = new QSettings( QStringLiteral( "QGIS" ), QStringLiteral( "QGISCUSTOMIZATION2" ) );
-  }
 
   // Using the customizationfile option always overrides the option and config path options.
   if ( !customizationfile.isEmpty() )
   {
     customizationsettings = new QSettings( customizationfile, QSettings::IniFormat );
     QgsCustomization::instance()->setEnabled( true );
   }
+  else
+  {
+    customizationsettings = new QSettings( QStringLiteral( "QGIS" ), QStringLiteral( "QGISCUSTOMIZATION2" ) );
+  }
 
   // Load and set possible default customization, must be done afterQgsApplication init and QgsSettings ( QCoreApplication ) init
   QgsCustomization::instance()->setSettings( customizationsettings );
```

Comment:
Not directly related to your work - but I've just noticed that customizationSettings is leaked here. (QgsCustomization is not deleting it)